.gitignore

```diff
@@ -23,3 +23,4 @@ testbin/*
 *.swp
 *.swo
 *~
+.vscode/*
```

Comment:
nit: how this relates to kustomize?

Reply:
it doesn't but my vscode decided to become verbose recently, and is outputting files and logs there. if you prefer I can drop that line from this PR

Reply:
going to split every change of this PR into mini tiny commits 
this is a nit PR mostly, cosmetic changes here and there

Reply:
Now every misselaneous change lives on it's own commit